nevergrad/benchmark/frozenexperiments.py

```diff
@@ -9,13 +9,53 @@
 from nevergrad.optimization.base import ConfiguredOptimizer
 from nevergrad.optimization import experimentalvariants  # pylint: disable=unused-import
 from nevergrad.functions import ArtificialFunction
+from nevergrad.functions.perfcap3d import core as perfcap3d
 from .xpbase import registry
 from .xpbase import create_seed_generator
 from .xpbase import Experiment
 
 # pylint: disable=stop-iteration-return, too-many-nested-blocks
 
 
+def perfcap_experiment(experiment_filename: str, seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
+    """
+    Factory function that creates Experiments for Perfcap3DFunction
+    This function is registered dynamically for 11 experiments under the names
+    'perfcap_bench1' to 'perfcap_bench11' in dynexpreg.py
+    """
+
+    seedg = create_seed_generator(seed)
+    budgets = [2000, 4000, 7000]
+    optimizer_names = [
+        "Shiwa",
+        "RandomSearch",
+        "RealSpacePSO",
+        "Powell",
+        "DiscreteOnePlusOne",
+        "CMA",
+        "NGO",
+        "TBPSA",
+        "chainCMAPowell",
+        "DE",
+    ]
+
+    # budgets = [10,20]
+    # optimizer_names = ["Shiwa","RandomSearch"]
+
+    total_experiment_count = len(budgets) * len(optimizer_names)
+    perfcap3d.Perfcap3DServerExecutor.execute_server(total_experiment_count)
+
+    # pylint: disable=stop-iteration-return
+    for budget in budgets:
+        for optim in optimizer_names:
+            yield Experiment(
+                function=perfcap3d.Perfcap3DFunction(experiment_filename),
```

Comment:
Implementation detail, please create the function in the outermost loop, it makes it much faster because the creation of the function can be costly compared to the rest of the loop.
In this case that means creating the function before the for loops, or if we merge the test cases into one big experiments, it means added `for experiment_filename in ....` as the outermost for loop, and create the corresponding function right after